rust/src/smb/smb2_records.rs

```diff
@@ -695,7 +695,7 @@ mod tests {
         let record: Smb2CreateRequestRecord = result.1;
         assert_eq!(record.disposition, 2); // FILE_CREATE: 2
         assert_eq!(record.create_options, 0x200021);
-        assert_eq!(record.data, &[]);
+        assert_eq!(record.data, &[] as &[u8]);
```

Comment:
What is happening here ?

Reply:
I think Serde introduces some trait impls that confuse the compiler to not knowing the `&[]`.  A better test might might be `assert!(record.data.is_empty())`.

Reply:
This looks a better test